### ISSUE TITLE
Allow usage of custom tags defined in native js libraries

### ISSUE THREAD

**Issue by sb-minimal** [ThoughtWorksInc/Binding.scala#43]
I read through the issues of custom tags but must admit, that i did not fully understand if it is possible to extend Binding.scala with custom tags.

Say  [OnsenUI](https://onsen.io) defines a custom tag `<ons-page>`, how can i use this in Binding.scala?

I tried the following wrapper
```
package onsen

import com.thoughtworks.binding._
import Binding._
import org.scalajs.dom.document
import org.scalajs.dom.raw.Node

import scala.language.implicitConversions
import scala.scalajs.js
import scala.scalajs.js.annotation.JSName

case class OnsenBasicComponentBuilder(tagName: String) {

  def render = this

  var modifier: Binding[String] = _

  @dom def build = {
    val element=document.createElement(tagName)
    element.setAttribute("modifier",modifier.bind)
    element
  }
}

object Onsen {
  implicit def autoBinding[A](a: A): Binding[A] = ???

  implicit final class CustomTags(x: dom.Runtime.TagsAndTags2.type) {

    def OnsPage() = OnsenBasicComponentBuilder("ons-page")
    def OnsToolbar() = OnsenBasicComponentBuilder("ons-toolbar")
    def OnsToolbarButton() = OnsenBasicComponentBuilder("ons-toolbar-button")
    def OnsIcon() = OnsenBasicComponentBuilder("ons-icon")
    def OnsButton() = OnsenBasicComponentBuilder("ons-button")

  }

  implicit def toHtml(x: OnsenBasicComponentBuilder): BindingSeq[Node] = {
    Constants(x.build).mapBinding(identity)
  }
}
```

and use it as follows
```
import onsen.Onsen._
  @dom
  def page = {
    <div>
      <OnsPage><OnsToolbar></OnsToolbar></OnsPage>
    </div>
  }
```

getting
```
overloaded method constructor NodeSeqMountPoint with alternatives:
[error]   (parent: org.scalajs.dom.raw.Node,childBinding: com.thoughtworks.binding.Binding[org.scalajs.dom.raw.Node])com.thoughtworks.binding.dom.Runtime.NodeSeqMountPoint <and>
[error]   (parent: org.scalajs.dom.raw.Node,childBinding: com.thoughtworks.binding.Binding[com.thoughtworks.binding.Binding.BindingSeq[org.scalajs.dom.raw.Node]],dummy: Unit)com.thoughtworks.binding.dom.Runtime.NodeSeqMountPoint <and>
[error]   (parent: org.scalajs.dom.raw.Node,childrenBinding: com.thoughtworks.binding.Binding.BindingSeq[org.scalajs.dom.raw.Node])com.thoughtworks.binding.dom.Runtime.NodeSeqMountPoint
[error]  cannot be applied to (onsen.OnsenBasicComponentBuilder, com.thoughtworks.binding.Binding.BindingSeq[org.scalajs.dom.raw.Node])
[error]       <OnsPage><OnsToolbar></OnsToolbar></OnsPage>
[error]        ^
```

**Comment by Atry** [comment]
I don't like tricks like this.
I would suggest you not to use custom tags. You should create simple custom functions instead.

**Comment by sb-minimal** [comment]
Ok, but how do i get Binding.Scala to emit `<ons-page><ons-toolbar></ons-toolbar></ons-page>` even with a "simple custom function" ?

**Comment by Atry** [comment]
Something like `onsPage(onsToolbar())`

**Comment by sb-minimal** [comment]
Yes, but how to i implement onsPage? It somehow has to produce Binding.Scala compatible tags of the kind "<ons-page>" on the browser.

**Comment by Atry** [comment]
Let your `onsPage` function return a `org.scalajs.dom.raw.Node`

**Comment by sadhen** [comment]
@sb-minimal 
It seems `<ons-page>${innerHTML}</ons-page>` is not supported.
But `<ons-page></ons-page>` is supported.

see my https://github.com/sadhen/Binding-SemanticUI for example

**Comment by Atry** [comment]
@sadhen , I guess @sb-minimal means custom **HTML** tags, which is not what you are talking about.

**Comment by sb-minimal** [comment]
guess I am getting something completely wrong.
I tried:
```
  def onsPage(content: BindingSeq[Node]) = {
    val element=document.createElement("ons-page")
    content.map{ c =>
      element.appendChild(c)
    }
    element
  }


  def onsToolbar(content: Node) = {
    document.createElement("ons-toolbar")
  }

  val pageTitle: Var[String]=Var("Titel")

  @dom
  def app = {
    <div> {
      onsPage(Seq(
        onsToolbar(
          <div class="left">back</div>
          <div class="center">{ pageTitle.get } </div>
        ),
        <p>
          content
        </p>
      ))
      }
    </div>
  }
```
getting 
```
type mismatch;
[error]  found   : com.thoughtworks.binding.Binding.BindingSeq[org.scalajs.dom.raw.Node]
[error]  required: org.scalajs.dom.raw.Node
[error]   @dom
[error]    ^
[error] one error found
```

Is `document.createElement("ons-page")` even the correct way to produce the `org.scalajs.dom.raw.Node`?
Would be very helpful if you could point me to some example code or could give a simple implementation for `onsPage()`.

**Comment by Atry** [comment]
You can use [NodeSeqMountPoint](https://static.javadoc.io/com.thoughtworks.binding/unidoc_2.11/10.0.2/com/thoughtworks/binding/dom$$Runtime$$NodeSeqMountPoint.html) at the moment, see https://scalafiddle.io/sf/gII6UlB/0

**Comment by Atry** [comment]
However, I guess it would be good if we have a special syntax to disable tag name checking, like https://github.com/ThoughtWorksInc/Binding.scala#custom-attributes

**Comment by sb-minimal** [comment]
i got it somewhat working using `NodeSeqMountPoint` as you suggested, but then ran into several conversion problems between `Node`, `Element`, `Binding` and `BindingSeq` and had to sometimes use the Elements directly, sometimes wrap into `Constants(...)`, and sometimes had to introduce a extra `<div>` wrapping my contents, ... which somewhat slowed down my productivity and made the library less intuitive to use.

Finally i ran into some compatibility issues when using `Binding.scala` together with `autowire` (seemed as some macro tried to redefine already existing helper functions).

So finally i switched to `scalajs-react` and got everything working within a few hours. Personally I'd appreciated to use `Binding.scala` and getting rid of Virtual DOM, but right now `scalajs-react` is so much easier to work with custom tags. 
A simple `val customTag   = "customTag"  .reactTag` there was enough for most of my needs.

For my currect project I'll stick with `scalajs-react`
 Maybe I'll look back at `Binding.scala` for the next project.

**Comment by Atry** [comment]
Fixed in #110
